ShareTheTable/View Controllers/ConversationViewController.m

```diff
@@ -56,18 +56,27 @@ - (void)refreshConversationData {
 }
 
 - (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
-    MessageCell* cell = [tableView dequeueReusableCellWithIdentifier:@"MessageCell" forIndexPath:indexPath];
     
     Message* message = self.messages[indexPath.row];
     [message fetch];
     
     PFUser* user = [PFUser objectWithoutDataWithObjectId:message.senderID];
     [user fetch];
     
-    cell.messageBodyText.text = message.messageBodyText;
-    cell.messageUserName.text = user.username;
-    
-    return cell;
+    // If message is from the current user logged in
+    if(PFUser.currentUser.username == user.username) {
+        MessageCell* cell = [tableView dequeueReusableCellWithIdentifier:@"MessageCell" forIndexPath:indexPath];
```

Comment:
You should probably name this Sender Message Cell or something similar to contrast with the Receiver Message Cell below